ext/AgentsVisualizations/src/utils.jl

```diff
@@ -1,6 +1,89 @@
 export subscript, superscript
 export record_interaction
 
+function Agents.check_space_visualization_API(::ABM{S}) where {S}
```

Comment:
I am not sure about this function. It is lots of code that is hard to maintain and does not really offer much. If a method is not implemented, the developer will get a `MethodError` error message when trying to run the code. We have already instructed them to go through the file and implement the methods in the file `space-visualization-API.jl`. I understand that it took you a lot of effort to write this function, but perhaps we should not keep it. It's lots of lines of code that offer little and would incur maintainance burder each time something new is added to the plotting API.

Reply:
Hm. I think this function is a big step forward in terms of developer experience. It provides an easy way to check if the own space type *can be plotted* and if something *optional is missing*.

I see your point though. It may prove a bit cumbersome in the long term to maintain this function. But I think the plotting API is pretty fixed for now and whenever something gets added to it, we just have to make it a requirement that the method check is added to this function. Not that hard to do imho but certainly a thing that has to be kept in mind.

Reply:
To me it seems a good idea instead

- not too hard to maintain
- cool
- what if a method is called rarely and the method error so happens unexpectedly? I think that knowing in advance the methods one should implement is a safeguard in this regard

Reply:
Okay, I accept, go ahead and resolve the thread!